src/java.desktop/share/classes/javax/swing/plaf/synth/SynthTreeUI.java

```diff
@@ -72,8 +72,16 @@ public class SynthTreeUI extends BasicTreeUI
     private int padding;
 
     private boolean useTreeColors;
+    private enum IconType {
+        EXPANDED,
+        COLLAPSED
+    }
+
+    private Icon expandedIconWrapper = new IconWrapper(IconType.EXPANDED);
+
+    private Icon collapsedIconWrapper = new IconWrapper(IconType.COLLAPSED);
```

Comment:
I guess you can pass in `collapsedIcon` or `expandedIcon`  to `IconWrapper `class which can save it in `icon` field and
Then IconWrapper class can just call `SynthGraphicsUtils.paintIcon(icon, context, g, x, y, w, h);`

Reply:
I guess those might be null at class initialisation stage so it didn't render..you can try 
```
expandedIconWrapper = new IconWrapper(expandedIcon);
collapsedIconWrappr - new IconWrapper(collapsedICon);
```
in updateStyle() where it sets
```
setExpandedIcon(style.getIcon(context, "Tree.expandedIcon"));
setCollapsedIcon(style.getIcon(context, "Tree.collapsedIcon"));
```

Reply:
Updated.